src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/__tests__/PipelineStageAdvancedSpecifications.test.tsx

```diff
@@ -0,0 +1,243 @@
+    userEvent.click(button as HTMLElement)
+
+    expect(pipelineContextMockValue.getStageFromPipeline).toBeCalled()
+    expect(pipelineContextMockValue.getStageFromPipeline).toBeCalledWith(
+      pipelineContextMockValue.state.selectionState.selectedStageId
+    )
+    expect(pipelineContextMockValue.updateStage).toBeCalled()
+
+    const conditionalExecutionHeader = screen.getByTestId('conditionalExecutionHeader')
+    const multiTypeButton = within(conditionalExecutionHeader).getByRole('button')
+    userEvent.click(multiTypeButton)
+    const runtimeInputText = screen.getByText('Runtime input')
+    await waitFor(() => expect(runtimeInputText).toBeInTheDocument())
+    userEvent.click(runtimeInputText)
+    expect(pipelineContextMockValue.updateStage).toBeCalled()
+
+    userEvent.click(multiTypeButton)
+    const fixedValueText = screen.getByText('Fixed value')
+    await waitFor(() => expect(fixedValueText).toBeInTheDocument())
+    userEvent.click(fixedValueText)
+    expect(pipelineContextMockValue.updateStage).toBeCalled()
+  })
+
+  test('should onUpdate FailureStrategies be called', async () => {
+    renderComponent()
+
+    const button = await waitFor(() => screen.findByText('Failure Strategy button'))
+    userEvent.click(button as HTMLElement)
+
+    expect(pipelineContextMockValue.getStageFromPipeline).toBeCalled()
+    expect(pipelineContextMockValue.getStageFromPipeline).toBeCalledWith(
+      pipelineContextMockValue.state.selectionState.selectedStageId
+    )
+    expect(pipelineContextMockValue.updateStage).toBeCalled()
+
+    const failureStrategyHeader = screen.getByTestId('failureStrategyHeader')
+    const multiTypeButton = within(failureStrategyHeader).getByRole('button')
+    userEvent.click(multiTypeButton)
+    const runtimeInputText = screen.getByText('Runtime input')
+    await waitFor(() => expect(runtimeInputText).toBeInTheDocument())
+    userEvent.click(runtimeInputText)
+    expect(pipelineContextMockValue.updateStage).toBeCalled()
+
+    userEvent.click(multiTypeButton)
+    const fixedValueText = screen.getByText('Fixed value')
+    await waitFor(() => expect(fixedValueText).toBeInTheDocument())
+    userEvent.click(fixedValueText)
+    expect(pipelineContextMockValue.updateStage).toBeCalled()
+  })
+
+  test('should onUpdate be called with undefined stage', async () => {
+    renderComponent({
+      ...pipelineContextMockValue,
+      getStageFromPipeline: jest.fn(() => {
+        return { stage: { stage: undefined }, parent: undefined }
+      }),
+      state: {
+        ...pipelineContextMockValue.state,
+        selectionState: {
+          selectedSectionId: 'ADVANCED',
+          selectedStageId: 'parStage1',
+          selectedStepId: undefined
+        }
+      }
+    })
+
+    const delegateSelectorButton = await waitFor(() =>
+      screen.findByPlaceholderText('delegate.Delegate_Selector_placeholder')
+    )
```

Comment:
unnecessary `waitFor`
```suggestion
    const delegateSelectorButton = await screen.findByPlaceholderText('delegate.Delegate_Selector_placeholder')
```